apps/network-alerts/js/attention/notify.js

```diff
@@ -0,0 +1,116 @@
+/* global
+  AudioContext
+ */
+
+(function(exports) {
+  'use strict';
+
+  const SETTINGS = {
+    notificationVolume: 'audio.volume.notification',
+    vibration: 'vibration.enabled'
+  };
+
+  const ATTENTION_PATTERN = [2000, 500, 1000, 500, 1000, 500];
+  ATTENTION_PATTERN.push(...ATTENTION_PATTERN);
+
+  const ATTENTION_SOUND_VOLUME = 0.3;
+
+  function getSetting(key) {
+    if (!navigator.mozSettings) {
+      return Promise.reject(new Error('The mozSettings API is not available.'));
+    }
+
+    return navigator.mozSettings.createLock().get(key).then(
+      result => result[key]
+    );
+  }
+
+  function getSettings(settings) {
+    return Promise.all(
+      settings.map(getSetting)
+    ).catch((e) => {
+      // catch and log errors
+      console.error('Error while retrieving settings', e.message, e);
+      return settings.map(() => null);
+    }).then((results) => {
+      return settings.reduce((result, setting, i) => {
+        result[setting] = results[i];
+        return result;
+      }, {});
+    });
+  }
+
+  // Converts from the ATTENTION_PATTERN (suitable for the Vibration API) to a
+  // Float32Array suitable for the Audio API.
+  function getAttentionCurveWave() {
```

Comment:
Interesting trick! It'd be great if you can add more detailed comment on how this method actually works :)